Automattic/wp-calypso
# Congrats: add business plan upsell banner

Related to https://github.com/Automattic/dotcom-forge/issues/4187

## Proposed Changes
ATM, if a user purchases the Explorer plan, we show the Creator upsell page (instead of the standard congrats page).
With this PR, we will render our standard congrats page, but at the end of the page will be an upsell banner.
The banner appears for purchasing any plan before Creator - for Starter and Explorer plans.

The design: VLoPHWqcewxNKZYjjkDu3O-fi-3148_2751

<table>
	<tr>
		<td> BEFORE
		<td> AFTER
	<tr>
	<tr>
		<td><img width="997" alt="Screenshot 2024-05-07 at 11 55 55" src="https://github.com/Automattic/wp-calypso/assets/5598437/7f3aeccf-bdd1-4b8c-b1f0-2436ad79d2ea">
		<td><img width="1056" alt="Screenshot 2024-05-07 at 11 56 34" src="https://github.com/Automattic/wp-calypso/assets/5598437/dbfe10f6-1c4e-474c-add3-af8974de6224">
	<tr>
</table>

## Testing Instructions
### Test Starter plan (monthly)
1) Let's create website with "Free plan" - open `/sites`, click on "Add new site", and proceed with setting up a site with `Free` plan
2) Open "Upgrades" in the left sidebar
3) Click on "Plans"
4) Crucial for current test case - at the top of the page select *"Pay monthly"*<br /><img width="298" alt="Screenshot 2024-05-06 at 14 16 40" src="https://github.com/Automattic/wp-calypso/assets/5598437/00c7f700-554b-4983-99b5-46b2b2382260">
5) Click on"Upgrade" button for "Starter" plan and proceed with purchasing
6) Assert that you see congrats page as previously, but at the bottom is located upsell banner with the next information: `7-day money-back` (since it's monthly, for yearly will be 14), and the Creator upgrade difference is €31 (since original price is 40, minus 9 for Started plan is already paid). <br /> <img width="687" alt="Screenshot 2024-05-06 at 14 10 22" src="https://github.com/Automattic/wp-calypso/assets/5598437/6b28fc98-c318-42d6-815a-09da684e45af">
7) Click on "Upgrade"
8) Assert that the info on checkout page is correct: price is 31 and subscription is monthly: <br /><img width="614" alt="Screenshot 2024-05-06 at 14 11 31" src="https://github.com/Automattic/wp-calypso/assets/5598437/0d44ba61-36aa-4bf1-a210-3a96686863df">

### Test Explorer plan (monthly)
1) Do the steps above, but purchase Explorer plan, instead of Starter. NB don't forget to select "Pay monthly"
2) Assert that everything works the same way: 7 days money back, upgrade price is 22 (44 - 18 for Explorer)

### Test Creator and Entrepreneur plans
1) Do the same steps as we have in the first case but buy Creator plan (additionally you can do the same with Entrepreneur)
2) Assert that you don't see the upsell banner

### Test Yearly subscription
1) Do the steps from the first case for Starter and then for Explorer. NB this time select "Pay yearly"
2) Assert that you see the upsell banner but this time money-back period is 14 days.

### Nice to test
Test `Pay every 2 years`, `Pay every 3 years` for every plan and assert that upsell banner is visible/hidden, original discount prices are correct inside banenr, on checkout page, and also money-back is correct (14 days for these cases).
